mathext/digamma.go

```diff
@@ -10,19 +10,35 @@ import (
 
 // Digamma returns the logorithmic derivative of the gamma function at x.
 //  ψ(x) = d/dx (Ln (Γ(x)).
-// Note that if x is a negative integer in [-7, 0] this function will return
-// negative Inf.
 func Digamma(x float64) float64 {
-	// This is adapted from
```

Comment:
Thanks for the extra citations. I still feel like we owe a reference to that code. It gives the algorithm as well as the specific values for the polynomial expansion.

Reply:
I also opened https://github.com/gonum/gonum/issues/302 for reference.